apps/api-extractor/src/generators/DeclarationReferenceGenerator.ts

```diff
@@ -99,68 +87,48 @@ export class DeclarationReferenceGenerator {
-        parent.globalExports &&
-        DeclarationReferenceGenerator._isSameSymbol(parent.globalExports.get(symbol.escapedName), symbol)
-      ) {
-        return 'global';
-      }
+
+    // If it's a global, then use an Exports navigation.
+    if (sourceFile && !ts.isExternalModule(sourceFile)) {
+      return Navigation.Exports;
     }
 
-    // Next, try determining navigation to symbol by its node
-    if (symbol.valueDeclaration) {
-      const declaration: ts.Declaration = ts.isBindingElement(symbol.valueDeclaration)
-        ? ts.walkUpBindingElementsAndPatterns(symbol.valueDeclaration)
-        : symbol.valueDeclaration;
-      if (ts.isClassElement(declaration) && ts.isClassLike(declaration.parent)) {
-        // class members are an "export" if they have the static modifier.
-        return ts.getCombinedModifierFlags(declaration) & ts.ModifierFlags.Static
-          ? Navigation.Exports
-          : Navigation.Members;
-      }
-      if (ts.isTypeElement(declaration) || ts.isObjectLiteralElement(declaration)) {
-        // type and object literal element members are just members
-        return Navigation.Members;
-      }
-      if (ts.isEnumMember(declaration)) {
-        // enum members are exports
-        return Navigation.Exports;
-      }
-      if (
-        ts.isExportSpecifier(declaration) ||
-        ts.isExportAssignment(declaration) ||
-        ts.isExportSpecifier(declaration) ||
-        ts.isExportDeclaration(declaration) ||
-        ts.isNamedExports(declaration)
-      ) {
-        return Navigation.Exports;
+    // If it's from an external library, then use an Exports navigation.
+    if (sourceFile && this._collector.program.isSourceFileFromExternalLibrary(sourceFile)) {
+      return Navigation.Exports;
+    }
+
+    // Otherwise, this symbol is from the current package.
+    if (parent) {
+      // If we've found an exported CollectorEntity, then it's exported from the package entry point, so
+      // use an Exports navigation.
+      const namedDeclaration: ts.DeclarationName | undefined = (
+        declaration as ts.NamedDeclaration | undefined
+      )?.name;
+      if (namedDeclaration && ts.isIdentifier(namedDeclaration)) {
+        const collectorEntity: CollectorEntity | undefined =
+          this._collector.tryGetEntityForNode(namedDeclaration);
+        if (collectorEntity && collectorEntity.exported) {
+          return Navigation.Exports;
+        }
       }
-      // declarations are exports if they have an `export` modifier.
-      if (ts.getCombinedModifierFlags(declaration) & ts.ModifierFlags.Export) {
+
+      // If its parent symbol is not a source file, then use an Exports navigation. If the parent symbol is
+      // a source file, but it wasn't exported from the package entry point (in the check above), then the symbol
+      // is a local, so fall through below.
+      if (!DeclarationReferenceGenerator._isExternalModuleSymbol(parent)) {
         return Navigation.Exports;
       }
```

Comment:
In my testing, I could not find a scenario in which any line from 112 to 157 was hit. I'm worried I may be missing an entire class of cases here... the original logic was added in the mega-PR https://github.com/microsoft/rushstack/pull/1337.

Reply:
@rbuckton any idea what kind of type declaration input would cover these these code paths?

Reply:
@octogonz shared some instances where `Members` navigation was returned, and I'm covering those now. But I still can not find a scenario where any line from 112-157 was hit.